TestUniqueIndexCreate_AddingDocWithNilValue_ShouldSucceed fails change detector

Strangely it looks like the index thinks there are multiple nil values on a change detector run:

```
=== RUN   TestUniqueIndexCreate_AddingDocWithNilValue_ShouldSucceed
        	            	2024-01-10T18:47:58.553Z, INFO, tests.integration, Test adding a doc with nil value for indexed field should succeed, {"database": "badger-file-system", "client": "go", "mutationType": "collection-save", "databaseDir": "", "changeDetector.Enabled": true, "changeDetector.SetupOnly": false, "changeDetector.SourceBranch": "develop", "changeDetector.TargetBranch": "", "changeDetector.Repository": "https://github.com/sourcenetwork/defradb.git"}
        	            	    utils2.go:1627: 
        	            	        	Error Trace:	/home/runner/work/defradb/defradb/tests/integration/utils2.go:1627
        	            	        	            				/home/runner/work/defradb/defradb/tests/integration/utils2.go:1098
        	            	        	            				/home/runner/work/defradb/defradb/tests/integration/utils2.go:277
        	            	        	            				/home/runner/work/defradb/defradb/tests/integration/utils2.go:206
        	            	        	            				/home/runner/work/defradb/defradb/tests/integration/utils2.go:162
        	            	        	            				/home/runner/work/defradb/defradb/tests/integration/index/create_unique_test.go:258
        	            	        	Error:      	Received unexpected error:
        	            	        	            	can not index a doc's field that violates unique index. DocID: bae-decf6467-4c7c-50d7-b09d-0a7097ef6bad, Field name: age, Field value: <nil>. Stack: /home/runner/work/defradb/defradb/db/errors.go:570 (0x1886758)
        	            	        	            		NewErrCanNotIndexNonUniqueField: return errors.New(
        	            	        	            	/home/runner/work/defradb/defradb/db/index.go:304 (0x1888173)
        	            	        	            		(*collectionUniqueIndex).newUniqueIndexError: return NewErrCanNotIndexNonUniqueField(doc.ID().String(), i.fieldDesc.Name, val)
        	            	        	            	/home/runner/work/defradb/defradb/db/index.go:322 (0x188833c)
        	            	        	            		(*collectionUniqueIndex).Update: return i.newUniqueIndexError(newDoc)
        	            	        	            	/home/runner/work/defradb/defradb/db/collection_index.go:136 (0x187b284)
        	            	        	            		(*collection).updateIndexedDoc: err = index.Update(ctx, txn, oldDoc, doc)
        	            	        	            	/home/runner/work/defradb/defradb/db/collection.go:945 (0x18748d3)
        	            	        	            		(*collection).save: err := c.updateIndexedDoc(ctx, txn, doc)
        	            	        	            	/home/runner/work/defradb/defradb/db/collection.go:900 (0x1874571)
        	            	        	            		(*collection).update: if err != nil {
        	            	        	            	/home/runner/work/defradb/defradb/db/collection.go:927 (0x187457b)
        	            	        	            		(*collection).Save: err = c.update(ctx, txn, doc)
        	            	        	            	/home/runner/work/defradb/defradb/tests/integration/utils2.go:1122 (0x18d3d75)
        	            	        	            		createDocViaColSave: return doc, collections[action.CollectionID].Save(s.ctx, doc)
        	            	        	            	/home/runner/work/defradb/defradb/tests/integration/utils2.go:1094 (0x18d3b6e)
        	            	        	            		createDoc.func1: doc, err = mutation(s, action, actionNodes[nodeID], collections)
        	            	        	            	/home/runner/work/defradb/defradb/tests/integration/utils2.go:1460 (0x18d6f71)
        	            	        	            		withRetry: err := action()
        	            	        	            	/home/runner/work/defradb/defradb/tests/integration/utils2.go:1089 (0x18d373a)
        	            	        	            		createDoc: err := withRetry(
        	            	        	            	/home/runner/work/defradb/defradb/tests/integration/utils2.go:277 (0x18cc0a8)
        	            	        	            		performAction: createDoc(s, action)
        	            	        	            	/home/runner/work/defradb/defradb/tests/integration/utils2.go:206 (0x18cbc10)
        	            	        	            		executeTestCase: performAction(s, i, testCase.Actions[i])
        	            	        	            	/home/runner/work/defradb/defradb/tests/integration/utils2.go:162 (0x18cb1e5)
        	            	        	            		ExecuteTestCase: executeTestCase(ctx, t, collectionNames, testCase, dbt, ct)
        	            	        	            	/home/runner/work/defradb/defradb/tests/integration/index/create_unique_test.go:258 (0x18e6665)
        	            	        	            		TestUniqueIndexCreate_AddingDocWithNilValue_ShouldSucceed: testUtils.ExecuteTestCase(t, test)
        	            	        	            	/opt/hostedtoolcache/go/1.20.12/x64/src/testing/testing.go:1576 (0x830eeb)
        	            	        	            		tRunner: fn(t)
        	            	        	            	/opt/hostedtoolcache/go/1.20.12/x64/src/runtime/asm_amd64.s:1598 (0x76a5e1)
        	            	        	            		goexit: BYTE	$0x90	// NOP
        	            	        	Test:       	TestUniqueIndexCreate_AddingDocWithNilValue_ShouldSucceed
        	            	        	Messages:   	Test adding a doc with nil value for indexed field should succeed
        	            	2024-01-10T18:47:58.589Z, INFO, db, Closing DefraDB process...
        	            	2024-01-10T18:47:58.592Z, INFO, db, Successfully closed running process
        	            	--- FAIL: TestUniqueIndexCreate_AddingDocWithNilValue_ShouldSucceed (0.04s)
```
https://github.com/sourcenetwork/defradb/actions/runs/7479363522/job/20356365281?pr=2185 and https://github.com/sourcenetwork/defradb/actions/runs/7476534873/job/20347123220

--- a/tests/integration/utils2.go
+++ b/tests/integration/utils2.go
@@ -186,7 +186,7 @@ func executeTestCase(
 		logging.NewKV("changeDetector.Repository", changeDetector.Repository),
 	)
 
-	startActionIndex, endActionIndex := getActionRange(testCase)
+	startActionIndex, endActionIndex := getActionRange(t, testCase)
 
 	s := newState(ctx, t, testCase, dbt, clientType, collectionNames)
 	setStartingNodes(s)
@@ -555,7 +555,7 @@ func flattenActions(testCase *TestCase) {
 //
 // If a SetupComplete action is provided, the actions will be split there, if not
 // they will be split at the first non SchemaUpdate/CreateDoc/UpdateDoc action.
-func getActionRange(testCase TestCase) (int, int) {
+func getActionRange(t *testing.T, testCase TestCase) (int, int) {
 	startIndex := 0
 	endIndex := len(testCase.Actions) - 1
 
@@ -598,8 +598,10 @@ ActionLoop:
 			// We must not set this to -1 :)
 			startIndex = firstNonSetupIndex
 		} else {
-			// if we don't have any non-mutation actions, just use the last action
-			startIndex = endIndex
+			// if we don't have any non-mutation actions and the change detector is enabled
+			// skip this test as we will not gain anything from running (change detector would
+			// run an idential profile to a normal test run)
+			t.Skipf("no actions to execute")
 		}
 	}
 

